### DISCUSSION TITLE
Fetching raw file from private GitHub repo fails with code 415

### DISCUSSION THREAD

**Discussion by nafpliot** [renovatebot/renovate#28912] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub

### Please tell us more about your question or problem

I am having a customDatasource to fetch a plaintext file (which contains a version number) from a private GitHub repository
```
    "customDatasources": {
      "rancher": {
        "defaultRegistryUrlTemplate": "https://api.github.com/repos/ORG/REPO/contents/k8s_version.txt",
        "format": "plain"
      }
```
However I get a 415 error (please see the logs). I assume this is because the default host rule for the `api.github.com` doesn't use the `application/vnd.github.v3.raw` header that is needed.

I have tried setting a hostRule to match the host and add my custom header (after I enabled it) like this
```
      {
        "matchHost": "https://api.github.com/repos/ORG/REPO/contents/k8s_version.txt",
        "token": "mytoken",
        "headers": {
          "Accept": "application/vnd.github.v3.raw"
        }
      }
```
but it doesn't work - I get the same error. I reckon than the hostRules don't override the default ones...

My question is, is there any way around it?

Thank you.

### Logs (if relevant)

<details><summary>Logs</summary>

```

DEBUG: GET https://api.github.com/repos/ORG/REPO/contents/k8s_version.txt = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=415 retryCount=0, duration=186) (repository=ORG/REPO)
```

</details>

**Comment by rarkins** [comment]
Can you reproduce the 415 with `curl`?

**Reply by nafpliot** [reply]
Hi @rarkins . No, with curl it works as expected. I get the json reply without the headers, or the normal plaintext with the headers.

**Comment by viceice** [comment, marked as answer]
Why not using `https://raw.githubusercontent.com/ORG/REPO/HEAD/k8s_version.txt` ?

**Reply by nafpliot** [reply]
ok... I was trying that without the https:// and it didn't work so I assumed that api requests to raw.githubusercontent.com didn't work. I was missing https:// . Thank you @viceice !
